src/meta/src/barrier/context/recovery.rs

```diff
@@ -468,6 +437,42 @@ impl GlobalBarrierWorkerContextImpl {
                     .await
                     .context("purge state table from hummock")?;
 
+                    let background_jobs = {
+                        let mut background_jobs = HashMap::new();
+
+                        let job_infos: HashMap<_, _> = info
+                            .values()
+                            .flatten()
+                            .filter(|(job_id, _)| {
+                                background_streaming_jobs.contains(&(job_id.table_id as _))
+                            })
+                            .map(|(table_id, job_info)| {
+                                (
+                                    table_id.table_id() as ObjectId,
+                                    job_info
+                                        .fragment_infos
+                                        .iter()
+                                        .map(|(fragment_id, fragment_info)| {
+                                            (*fragment_id as _, fragment_info.clone())
+                                        })
+                                        .collect(),
+                                )
+                            })
+                            .collect();
+                        let jobs = self
+                            .metadata_manager
+                            .catalog_controller
+                            .restore_background_streaming_job(job_infos)
+                            .await?;
+
+                        for (job_id, info) in jobs {
```

Comment:
Previously we have logic to call `finish_streaming_job` for jobs with no backfill actors to track. Do we need to preserve this logic?